test/e2e/minscale_readiness_test.go

```diff
@@ -138,15 +141,31 @@ func latestRevisionName(t *testing.T, clients *test.Clients, configName string)
 	return config.Status.LatestCreatedRevisionName
 }
 
-func waitForDesiredScale(t *testing.T, clients *test.Clients, deploymentName string, cond func(int32) bool) error {
-	deployments := clients.KubeClient.Kube.AppsV1().Deployments(test.ServingNamespace)
+func serverlessServicesName(t *testing.T, clients *test.Clients, revisionName string) string {
+	if err := wait.PollImmediate(time.Second, 10*time.Second, func() (bool, error) {
+		_, err := clients.NetworkingClient.ServerlessServices.Get(revisionName, metav1.GetOptions{})
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		t.Fatalf("Error retrieving sks %q: %v", revisionName, err)
+	}
+	sks, err := clients.NetworkingClient.ServerlessServices.Get(revisionName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to get sks after it was seen to be live: %v", err)
+	}
+	return sks.Status.PrivateServiceName
```

Comment:
Is this guaranteed to be set? 🤔 AFAIK we need to go through at least one reconcilation cycle to set this. I'd propose to cache it from the above like so:

```golang
func serverlessServicesName(t *testing.T, clients *test.Clients, revisionName string) string {
	var privateServiceName string
	if err := wait.PollImmediate(time.Second, 1*time.Minute, func() (bool, error) {
		_, err := clients.NetworkingClient.ServerlessServices.Get(revisionName, metav1.GetOptions{})
		if err != nil {
			return false, nil
		}
		
		privateServiceName = sks.Status.PrivateServiceName
		if privateServiceName == "" {
			return false, nil
		} 
		return true, nil
	}); err != nil {
		t.Fatalf("Error retrieving sks %q: %v", revisionName, err)
	}
	return privateServiceName
}
```

That should make this safe against potential inconsistencies. I'd also bump the poll timeout to 1 minute here probably.